pkg/serviceprovider/gitlab/downloadfilecapability_test.go

```diff
@@ -69,7 +69,10 @@ func TestGetFileHead(t *testing.T) {
 		tokenStorage: ts,
 	}
 
-	fileCapability := NewDownloadFileCapability(client, gitlabClientBuilder, "https://fake.github.com")
+	repoUrlMatcher, err := newRepoUrlMatcher("https://fake.github.com")
```

Comment:
Yes, the reason for that is we are setting the fakeRoundTripper with the `https://fake.gitlab.com` URL here: https://github.com/redhat-appstudio/service-provider-integration-operator/pull/598/files#diff-0c8448138fb76f7a0df72acf039643b775099efdcab8742453a9715b93d69f39R43

The baseUrl set for downloadfilecapability has to match it otherwise the mocked response is not returned.
Of course, I could have changed it but I didn't want to make unnecessary changes.